hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieInstantTimeGenerator.java

```diff
@@ -152,6 +152,15 @@ public static String formatDate(Date timestamp) {
     return getInstantFromTemporalAccessor(convertDateToTemporalAccessor(timestamp));
   }
 
+  public static String formatDateBasedOnTimeZone(Date timestamp) {
+    if (HoodieTimelineTimeZone.UTC.equals(commitTimeZone)) {
+      return timestamp.toInstant().atZone(HoodieTimelineTimeZone.UTC.getZoneId())
+          .toLocalDateTime().format(MILLIS_INSTANT_TIME_FORMATTER);
+    } else {
+      return MILLIS_INSTANT_TIME_FORMATTER.format(convertDateToTemporalAccessor(timestamp));
```

Comment:
should we always use `commitTimeZone`? The `convertDateToTemporalAccessor` uses `ZoneId.systemDefault()` instead of the configured one.

Reply:
Let's revert it first and fire another ticket to address the timezone issue for `#createNewInstantTime`.